receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -122,13 +120,50 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 				ghs.logger.Sugar().Errorf("error getting contributor count for repo %s", zap.Error(err), repo.Name)
 			}
 			ghs.mb.RecordGitRepositoryContributorCountDataPoint(now, int64(contribs), name)
+
+			// Get Pull Request data
+			prs, err := ghs.getPullRequests(ctx, genClient, name)
+			if err != nil {
+				ghs.logger.Sugar().Errorf("error getting pull requests for repo %s", zap.Error(err), repo.Name)
+			}
+
+			var merged int
+			var open int
+
+			for _, pr := range prs {
+				if pr.Merged {
+					merged++
+
+					age := getAge(pr.CreatedAt, pr.MergedAt)
+
+					ghs.mb.RecordGitRepositoryPullRequestMergedTimeDataPoint(now, age, name, pr.HeadRefName)
+
+				} else {
+					open++
+
+					age := getAge(pr.CreatedAt, now.AsTime())
+
+					ghs.mb.RecordGitRepositoryPullRequestOpenTimeDataPoint(now, age, name, pr.HeadRefName)
```

Comment:
So depending on the pull request's status (merged or not), some metrics are emitted and some are not:
- `git.repository.pull_request.merged.time` is only emitted if the pull request is merged - I guess this makes sense 😄 
- `git.repository.pull_request.open.time` is only emitted if the pull request is open - I'm not sure about this
- `git.repository.pull_request.approved.time` is only emitted if the pull request is approved and still open - I'm not sure about this either.

Anyway, this component is in `development` stability, so I guess we should just merge, evaluate how this can be used, and adjust accordingly.

Can you describe the way(s) you intend to use (or are using) these metrics? That would help a lot reasoning what makes sense.

Reply:
> - `git.repository.pull_request.open.time` is only emitted if the pull request is open - I'm not sure about this
> - `git.repository.pull_request.approved.time` is only emitted if the pull request is approved and still open - I'm not sure about this either.

@adrielp and I talked about these when we redid the metrics last week and made this a conscious choice, but now I'm leaning more toward your take that it might be useful to emit no matter what.

Reply:
Understanding how these metrics are/will be used would help a lot here.

Reply:
Sorry for the delay in response to this, I've been away for the holidays and completely mentally checked out :D 


I like the comment about using state as an attribute and having these all be under the `pull_request` namespace. 

Time here really means `age`, but time is preferred name in the specification which is why we chose it.

Merged, open, and approved are all state, and thus when they change, they're essentially a different metric.

These metrics are used to help teams understand where their bottle necks are and how they could improve if they so choose.  With pull request metrics for example, a team would be able to understand what their lead time is for getting pull requests and contributions through the their code base.

If they have a lot of pull requests in the open state, and the time of which they've been open is long, then it's very likely the team is not pairing, or they have to few people available for PR reviews. 

And that might be ok for a team. But it also might be an indicator that processes should change. They're also not metrics to be used in isolation. They're meant to be used with the other metrics to form a picture.  

We have a whole doc on leading indicators and how teams could leverage these metrics that we've been working on publishing publicly. My intent (which hasn't made it in here yet) is to link & reference that guidance on these metrics within this receiver. Additionally, I'd include a link to the demo that runs and displays the visualization of these metrics.

I'd vote for merging as-is (we know these metrics work at this time in point) and make quick small changes to the names in subsequent PR's while in development.

Reply:
Awesome, I love your answer @adrielp , thank you for this.

> Sorry for the delay in response to this, I've been away for the holidays and completely mentally checked out 

That's completely OK and the right thing to do!

> We have a whole doc on leading indicators and how teams could leverage these metrics that we've been working on publishing

This sounds great, I definitely want to read it when it's public.

> I'd vote for merging as-is (we know these metrics work at this time in point) and make quick small changes

Makes sense to me, again - given that the component is in Development stability status, I suppose that's the way to go.

One last thing I'm waiting for (after the holiday season ends for everybody 🙂) is to update the README, I'm thinking about the GitHub Metrics section.﻿
![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/70892616/5e5a8f9a-5bb0-4a3f-b17a-7701223dca8b)

Reply:
Sounds good to me! Agreed on updating that readme. Hopefully that can be updated tomorrow & we can get this merged this so we can get some small pr's open to iterate! Appreciate the thorough review on this and insight!

Reply:
There we go, I just updated the README to fix this.